network/network_linux.go

```diff
@@ -187,6 +254,182 @@ func (nm *networkManager) deleteBridgeRules(extIf *externalInterface) {
+	hostIf, err := net.InterfaceByName(extIf.Name)
+	if err != nil {
+		return err
+	}
+
+	// If a bridge name is not specified, generate one based on the external interface index.
+	bridgeName := nwInfo.BridgeName
+	if bridgeName == "" {
+		bridgeName = fmt.Sprintf("%s%d", bridgePrefix, hostIf.Index)
+	}
+
+	// Check if the bridge already exists.
+	bridge, err := net.InterfaceByName(bridgeName)
+	if err != nil {
+		// Create the bridge.
+		log.Printf("[net] Creating bridge %v.", bridgeName)
+
+		err = nm.createOVSBridge(bridgeName)
+		if err != nil {
+			return err
+		}
+
+		// On failure, delete the bridge.
+		defer func() {
+			if err != nil {
+				nm.deleteOVSBridge(bridgeName)
+			}
+		}()
+
+		bridge, err = net.InterfaceByName(bridgeName)
+		if err != nil {
+			return err
+		}
+	} else {
+		// Use the existing bridge.
+		log.Printf("[net] Found existing bridge %v.", bridgeName)
+	}
+
+	// Save host IP configuration.
+	err = nm.saveIPConfig(hostIf, extIf)
+	if err != nil {
+		log.Printf("[net] Failed to save IP configuration for interface %v: %v.", hostIf.Name, err)
+	}
+
+	// External interface down.
+	log.Printf("[net] Setting link %v state down.", hostIf.Name)
+	err = netlink.SetLinkState(hostIf.Name, false)
+	if err != nil {
+		return err
+	}
+
+	// Connect the external interface to the bridge.
+	log.Printf("[net] Setting link %v master %v.", hostIf.Name, bridgeName)
+	err = setOVSMaster(hostIf.Name, bridgeName)
+	if err != nil {
+		return err
+	}
+
+	// Add the bridge rules.
+	err = nm.addOVSRules(extIf, hostIf, bridgeName)
+	if err != nil {
+		return err
+	}
+
+	// External interface up.
+	log.Printf("[net] Setting link %v state up.", hostIf.Name)
+	err = netlink.SetLinkState(hostIf.Name, true)
+	if err != nil {
+		return err
+	}
+
+	// Bridge up.
+	log.Printf("[net] Setting link %v state up.", bridgeName)
+	err = netlink.SetLinkState(bridgeName, true)
+	if err != nil {
+		return err
+	}
+
+	// Apply IP configuration to the bridge for host traffic.
+	err = nm.applyIPConfig(extIf, bridge)
+	if err != nil {
+		log.Printf("[net] Failed to apply interface IP configuration: %v.", err)
+	}
+
+	extIf.BridgeName = bridgeName
+	err = nil
+
+	log.Printf("[net] Connected interface %v to bridge %v.", extIf.Name, extIf.BridgeName)
+
+	return nil
+}
+
+// DisconnectExternalInterface disconnects a host interface from OVS bridge.
+func (nm *networkManager) disconnectOVSInterface(extIf *externalInterface) error {
+	log.Printf("[net] Disconnecting interface %v.", extIf.Name)
+
+	// Disconnect external interface from its bridge.
+	cmd := fmt.Sprintf("ovs-vsctl del-port %s %s", extIf.BridgeName, extIf.Name)
+	_, err := common.ExecuteShellCommand(cmd)
+	if err != nil {
+		log.Printf("[net] Failed to disconnect interface %v from bridge, err:%v.", extIf.Name, err)
+	}
+
+	// Delete the bridge.
+	err = nm.deleteOVSBridge(extIf.BridgeName)
+	if err != nil {
+		log.Printf("[net] Failed to delete bridge %v, err:%v.", extIf.BridgeName, err)
```

Comment:
Should we return err here?

Reply:
not needed. if it returns error, then we assume port already removed

Reply:
Then we always return nil. How can caller function catch such error?

Reply:
If something went wrong, we have to check logs. I didn't change any behaviour. I followed the same as that of linux bridge. If we change this, im not sure what we gonna break